src/software/new_geom/polygon.cpp

```diff
@@ -64,6 +64,18 @@ const std::vector<Point>& Polygon::getPoints() const
     return points_;
 }
 
+bool Polygon::isDegenerate()
+{
+    Point first_point = points_[0];
+    auto it =
+        std::find_if_not(points_.begin(), points_.end(),
+                         [first_point](const Point& p) { return first_point == p; });
```

Comment:
This took a second to wrap my head around, could we leave a quick comment above:
```
// Check that every point is the same as the first point
```

Reply:
done